src/test/regress/expected/multi_subquery_complex_queries.out

```diff
@@ -0,0 +1,870 @@
+SELECT *
+FROM
+  (SELECT "some_users_data".user_id, lastseen
+   FROM
+     (SELECT 2 * user_id as user_id,
+             (TIME) AS lastseen
+      FROM
+        (SELECT user_id,
+                TIME
+         FROM
+           (SELECT user_id,
+                   TIME
+            FROM events_table as "events"
+            WHERE user_id > 10 and user_id < 40) "events_1"
+         ORDER BY TIME DESC
+         LIMIT 1000) "recent_events_1"
+      ) "some_recent_users"
+   JOIN LATERAL
+     (SELECT "users".user_id
+      FROM users_table as "users"
+      WHERE "users"."user_id" = "some_recent_users"."user_id"
+        AND users.value_2 > 50 and users.value_2 < 55
+      LIMIT 1) "some_users_data" ON TRUE
+   ORDER BY lastseen DESC
+   LIMIT 50) "some_users"
+     order BY user_id
+     limit 50;
+ERROR:  cannot pushdown the subquery since all relations are not joined using distribution keys
+DETAIL:  Each relation should be joined with at least one another relation using distribution keys and equality operator.
+-- LATERAL JOINs used with INNER JOINs
+SELECT user_id, lastseen
+FROM
+  (SELECT "some_users_data".user_id, lastseen
+   FROM
+     (SELECT filter_users_1.user_id,
+             TIME AS lastseen
+      FROM
+        (SELECT user_where_1_1.user_id
+         FROM
+           (SELECT "users"."user_id"
+            FROM users_table as "users"
+            WHERE user_id > 12 and user_id < 16 and value_1 > 20) user_where_1_1
+         INNER JOIN
+           (SELECT "users"."user_id"
+            FROM users_table as "users"
+            WHERE   user_id > 12 and user_id < 16 and value_2 > 60) user_where_1_join_1 
+           ON ("user_where_1_1".user_id = "user_where_1_join_1".user_id)) filter_users_1 
+      JOIN LATERAL
+        (SELECT user_id,
+                TIME
+         FROM events_table as "events"
+         WHERE  user_id > 12 and user_id < 16 and user_id = filter_users_1.user_id
+         ORDER BY TIME DESC
+         LIMIT 1) "last_events_1" ON TRUE
+      ORDER BY TIME DESC
+      LIMIT 10) "some_recent_users"
+   JOIN LATERAL
+     (SELECT "users".user_id
+      FROM users_table  as "users"
+      WHERE "users"."user_id" = "some_recent_users"."user_id"
+             AND "users"."value_2" > 70
+      LIMIT 1) "some_users_data" ON TRUE
+   ORDER BY lastseen DESC
+   LIMIT 10) "some_users"
+order BY user_id DESC
+limit 10;
+ user_id |            lastseen             
+---------+---------------------------------
+      15 | Sat Mar 11 14:43:29.525317 2017
+      15 | Sat Mar 11 14:43:29.525317 2017
+      15 | Sat Mar 11 14:43:29.525317 2017
+      15 | Sat Mar 11 14:43:29.525317 2017
+      15 | Sat Mar 11 14:43:29.525317 2017
+      15 | Sat Mar 11 14:43:29.525317 2017
+      15 | Sat Mar 11 14:43:29.525317 2017
+      15 | Sat Mar 11 14:43:29.525317 2017
+      15 | Sat Mar 11 14:43:29.525317 2017
+      15 | Sat Mar 11 14:43:29.525317 2017
+(10 rows)
+
+-- not supported since the inner JOIN is not equi join
```

Comment:
> For failing use-cases, shall we use simpler queries? It would be easier to read regression tests and understand why they are failing.

I had comments on each test for why it is failing.

> I'm ok to keep current ones if updating them will take too much time. We can add an issue to think about them later.

I'm inclined to keep these complex queries as failure tests as well. One would only read the test if it somehow breaks it, so, considering the frequency of happening that, it should be fine to keep these longer tests 😄